### ISSUE TITLE
Request: appropriate error codes on success / failure

### ISSUE THREAD

**Issue by greghuc** [netlify/netlifyctl#18]
Hi there

Apparently this netlify go client will be the successor to the node one.

Could you please ensure that this client returns appropriate (unix) error codes, depending on success / failure? E.g. on website deploy success, return 0 else non-zero.

This makes it easier to integrate the netlify client into a large shell script.

Many thanks!

Greg

**Comment by calavera** [comment]
Thanks for opening this issue.

Yup, it should not return 0 when there is an error. Pull requests really welcome, but we still need to work on errors cases a lot.

**Comment by calavera** [comment]
@greghuc I just checked and all commands should exit with `-1` when there is any errors. Did you find any case where that wasn't the output for an error?

**Comment by greghuc** [comment]
I haven't found any cases where errors returned 0. However I raised this request to ensure the correct behaviour was specifically designed for and tested.

At present, I don't entirely trust the go-client as it's work-in-progress and has no tests. I have also seen the netlify node-client return wrong error codes, so I wanted to be sure that the go-client didn't. 

To get round (possibly) wrong error codes, I'm currently wrapping the netlifyctl command in my own shell script that definitely returns correct error codes: I set error codes based on whether the string "your website is live" appears in the netlifyctl output. I think having netlifyctl return a unique string on success is a nice fallback, in additional to correct error codes.

```
netlifyctl deploy --base-directory ${BUILD_DIR} --access-token ${NETLIFY_TOKEN} --config ${CONFIG_FILE} 2>&1 | tee -a ${DEPLOY_OUTPUT}
cat ${DEPLOY_OUTPUT} | grep -q 'your website is live'
DEPLOY_RESULT=$?
rm ${DEPLOY_OUTPUT}
if [ ${DEPLOY_RESULT} -eq 0 ]
then
  echo "Successfully deployed website from folder: ${BUILD_DIR}" && exit 0
else
  echo "Failed to deploy website from folder: ${BUILD_DIR}" && exit 1
fi
```
